jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocDriverUtils.java

```diff
@@ -0,0 +1,246 @@
+            abstract Builder setHost(String value);
+
+            abstract Builder setPort(int value);
+
+            abstract Builder setDbName(String value);
+
+            abstract Builder setHttpPath(String value);
+
+            abstract Builder setTransportMode(String value);
+
+            abstract Builder setOtherSessionConfs(String value);
+
+            abstract Builder setHiveConfs(String value);
+
+            abstract Builder setHiveVars(String value);
+
+            abstract HiveJdbcConnectionParam build();
+        }
+    }
+
+    /**
+     * Parses client url and extracts Hive connection parameters.
+     *
+     * @param url possible formats:
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;port=<>
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;clusterPoolLabel=foo=bar:env=staging;
+     * @return HiveJdbcConnectionParam object with fields correctly set
+     * @throws IOException
+     * @throws InvalidURLException
+     */
+    public static HiveJdbcConnectionParam parseHiveUrl(String url) throws InvalidURLException {
+        HiveJdbcConnectionParam.Builder paramBuilder = HiveJdbcConnectionParam.builder();
+        String urlStripped = url.substring(URL_HIVE_PREFIX.length());
+
+        String dbName = urlStripped.substring(0, urlStripped.indexOf(';'));
+        paramBuilder.setDbName(dbName.isEmpty() ? DEFAULT_DATABASE : dbName);
+
+        int hiveConfsStart = urlStripped.indexOf('?');
+        int hiveVarsStart = urlStripped.indexOf("#");
+
+        String hiveConfs =
+                hiveConfsStart == -1
+                        ? null
+                        : urlStripped.substring(
+                        hiveConfsStart + 1, hiveVarsStart == -1 ? urlStripped.length() : hiveVarsStart);
+        String hiveVars = hiveVarsStart == -1 ? null : urlStripped.substring(hiveVarsStart + 1);
+
+        // Params that do not need parsing
+        paramBuilder.setHiveConfs(hiveConfs);
+        paramBuilder.setHiveVars(hiveVars);
+
+        // Parse sessionConfigs to get cluster info
+        String sessionConfigs = urlStripped;
+        if (hiveConfsStart != -1 || hiveVarsStart != -1) {
+            sessionConfigs =
+                    urlStripped.substring(0, hiveConfsStart != -1 ? hiveConfsStart : hiveVarsStart);
+        }
+
+        // key=value pattern
+        Pattern pattern = Pattern.compile("([^;]*)=([^;]*)[;]?");
+        Matcher matcher = pattern.matcher(sessionConfigs);
+        StringBuilder sb = new StringBuilder();
+        String projectId = null;
+        String region = null;
+        String clusterPoolLabel = null;
+        String clusterName = null;
+        int port = DEFAULT_PORT;
+        while (matcher.find()) {
+            String pair = urlStripped.substring(matcher.start(), matcher.end()).replace(";", "");
+            String field = pair.split("=")[0];
+            String value = pair.split("=")[1];
```

Comment:
You should add a test that a reasonable exception is thrown if the user omits the equal sign like in:

```
jdbc:dataproc://hive/dbname;foo;bar;baz;
```

In that case, you probably want to throw InvalidUrlException